usaspending_api/download/management/commands/download_sqs_worker.py

```diff
@@ -1,110 +1,219 @@
                     time.sleep(1)
 
                 # If this process is exiting, don't poll for more work
                 keep_polling = not dispatcher.is_exiting
 
 
 def download_service_app(download_job_id):
+
+    download_job = _retrieve_download_job_from_db(download_job_id)
+    download_job_details = download_job_to_log_dict(download_job)
+
     with SubprocessTrace(
-        name=f"job.{JOB_TYPE}.download",
+        name=f"job.{JOB_TYPE}.download.start",
+        kind=SpanKind.INTERNAL,
         service="bulk-download",
-        span_type=SpanTypes.WORKER,
     ) as span:
-        download_job = _retrieve_download_job_from_db(download_job_id)
-        download_job_details = download_job_to_log_dict(download_job)
+        span.set_attributes(
+            {
+                "message": "Starting processing of download request",
+                "service": "bulk-download",
+                "span_type": str(SpanKind.INTERNAL),
+                "job_type": str(JOB_TYPE),
+                # download job details
+                "download_job_id": str(download_job_id),
+                "download_job_status": str(download_job.job_status.name),  # Convert to relevant field as str
+                "download_file_name": str(download_job.file_name),  # Extract specific field as str
+                "download_file_size": (
+                    download_job.file_size if download_job.file_size is not None else 0
+                ),  # int or fallback to 0
+                "number_of_rows": download_job.number_of_rows if download_job.number_of_rows is not None else 0,
+                "number_of_columns": (
+                    download_job.number_of_columns if download_job.number_of_columns is not None else 0
+                ),
+                "error_message": download_job.error_message if download_job.error_message else "",
+                "monthly_download": str(download_job.monthly_download),  # Convert boolean to str
+                "json_request": str(download_job.json_request) if download_job.json_request else "",
+            }
+        )
+
         log_job_message(
             logger=logger,
             message="Starting processing of download request",
             job_type=JOB_TYPE,
             job_id=download_job_id,
             other_params=download_job_details,
         )
-        span.set_tags(download_job_details)
+
         generate_download(download_job=download_job)
 
 
 def _retrieve_download_job_from_db(download_job_id):
+
     download_job = DownloadJob.objects.filter(download_job_id=download_job_id).first()
-    if download_job is None:
-        raise DownloadJobNoneError(download_job_id)
+    download_job_details = download_job_to_log_dict(download_job)
+
+    with SubprocessTrace(
+        name=f"job.{JOB_TYPE}.download.retrieve",
+        kind=SpanKind.INTERNAL,
+        service="bulk-download",
+    ) as span:
+
+        span.set_attributes(
```

Comment:
Seems like this could be broken out into a method since it is such a long list of attributes that get set and it appears to be used in multiple places. Won't block over this, but a suggestion to help with maintenance.

Reply:
I broke a few of these out. Let me know what you think. Might end up taking the logging back out of that new function